src/file-picker/src/FilePicker.js

```diff
@@ -47,6 +47,20 @@ export default class FilePicker extends PureComponent {
      */
     height: PropTypes.number,
 
+    /**
+     * The placeholder attribute of the input.
+     */
+    placeholder: PropTypes.string,
+
+    /**
+     * Button labels in different states of the component.
+     */
+    buttonLabels: PropTypes.shape({
```

Comment:
I think we could make all of these top level props instead and simplify some of the logic below.